src/H5Odtype.c

```diff
@@ -830,6 +830,44 @@ H5O__dtype_decode_helper(unsigned *ioflags /*in,out*/, const uint8_t **pp, H5T_t
                 dt->shared->force_conv = true;
             break;
 
+        case H5T_COMPLEX:
+            /*
+             * Complex number datatypes...
+             */
+
+            /* Set whether the complex number datatype is homogeneous */
+            dt->shared->u.cplx.homogeneous = (bool)(flags & 0x01);
+
+            if (!dt->shared->u.cplx.homogeneous)
+                HGOTO_ERROR(H5E_DATATYPE, H5E_UNSUPPORTED, FAIL,
+                            "heterogeneous complex number datatypes are currently unsupported");
+
+            /* Set the form of the complex number datatype */
+            dt->shared->u.cplx.form = (H5T_complex_form_t)((flags >> 1) & 0x03);
```

Comment:
From previous discussions, use the next two bits to store the type of the complex number. Currently, only rectangular form is supported.